hook!-full-game/src/Player/Player.gd

```diff
@@ -27,9 +27,15 @@ var info_dict: = {} setget set_info_dict
 
 
 func _ready() -> void:
+	Events.connect("level_loaded", self, "_on_Events_level_loaded")
 	stats.connect("health_depleted", state_machine, "transition_to", ['Die'])
 
 
+func _on_Events_level_loaded(level: Node2D) -> void:
+	var spawn_checkpoint = level.get_node("Checkpoints").get_child(0)
```

Comment:
The player shouldn't access the checkpoint like that imo. Guilherme has worked on a checkpoints node that keeps track of visited checkpoints, we should go through something like that. Although maybe it's a job for another task or PR

Reply:
Yeah I agree, I think this operation can be moved in `LevelLoader`, it should know how to do player setup/positioning as well